projects/plugins/jetpack/extensions/plugins/publicize/components/form/index.js

```diff
@@ -1,69 +1,55 @@
-
 /**
  * Internal dependencies
  */
-import PublicizeFormUnwrapped from '../form-unwrapped';
+import PublicizeConnection from '../connection';
+import PublicizeSettingsButton from '../settings-button';
+import MessageBoxControl from '../message-box-control';
+import useSocialMediaConnections from '../../hooks/use-social-media-connections';
+import useSocialMediaMessage from '../../hooks/use-social-media-message';
+
+export default function PublicizeForm( { refreshCallback } ) {
+	const { connections } = useSocialMediaConnections();
+	const { message, updateMessage, maxLength } = useSocialMediaMessage();
+	const { toggleEnableState } = useSocialMediaConnections();
 
-const PublicizeForm = compose( [
-	withSelect( select => {
-		return {
-			connections: select( 'core/editor' ).getEditedPostAttribute(
-				'jetpack_publicize_connections'
-			),
-			shareMessage: select( 'jetpack/publicize' ).getShareMessage(),
-			maxLength: select( 'jetpack/publicize' ).getShareMessageMaxLength(),
-		};
-	} ),
-	withDispatch( ( dispatch, { connections } ) => ( {
-		/**
-		 * Toggle connection enable/disable state based on checkbox.
-		 *
-		 * Saves enable/disable value to connections property in editor
-		 * in field 'jetpack_publicize_connections'.
-		 *
-		 * @param {number}  id - ID of the connection being enabled/disabled
-		 */
-		toggleConnection( id ) {
-			const newConnections = connections.map( connection => ( {
-				...connection,
-				enabled: connection.id === id ? ! connection.enabled : connection.enabled,
-			} ) );
+	function isDisabled() {
+		return connections.every( connection => ! connection.toggleable );
+	}
 
-			dispatch( 'core/editor' ).editPost( {
-				jetpack_publicize_connections: newConnections,
-			} );
-		},
+	return (
+		<div id="publicize-form">
+			<ul className="jetpack-publicize__connections-list">
+				{ connections.map( ( { display_name, enabled, id, service_name, toggleable } ) => (
+					<PublicizeConnection
+						disabled={ ! toggleable }
+						enabled={ enabled }
+						key={ id }
+						id={ id }
+						label={ display_name }
+						name={ service_name }
+						toggleConnection={ toggleEnableState }
+					/>
+				) ) }
+			</ul>
 
-		/**
-		 * Handler for when sharing message is edited.
-		 *
-		 * Saves edited message to state and to the editor
-		 * in field 'jetpack_publicize_message'.
-		 *
-		 * @param {object} message                - Message to change.
-		 * @param {boolean} hasEditedShareMessage - Whether the share message has been edited.
-		 */
-		messageChange( message, hasEditedShareMessage ) {
-			dispatch( 'core/editor' ).editPost( {
-				meta: {
-					jetpack_publicize_message: message,
-					jetpack_publicize_hasEditedShareMessage: hasEditedShareMessage,
-				},
-			} );
-		},
-	} ) ),
-] )( PublicizeFormUnwrapped );
+			<PublicizeSettingsButton refreshCallback={ refreshCallback } />
 
-export default PublicizeForm;
+			{ connections.some( connection => connection.enabled ) && (
```

Comment:
```suggestion
			{ connections.some( connection => connection.toggleable && connection.enabled ) && (
```

I'm not sure if this is the behaviour we want, but it would mean the message box isn't displayed at all if we don't have a connection that can be shared to.

Reply:
I think that's correct, and the current behavior in prod. This PR is mainly a refactoring process without introducing a big change in terms of its functionality. Is this suggestion something that we really wanna do now, here?